Sources/QuickTests/Helpers/XCTestCaseProvider.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+import XCTest
+
+// XCTestCaseProvider is defined on Linux as part of swift-corelibs-xctest,
+// but is not available on OS X. By defining this protocol on OS X, we ensure
+// that the tests fail on OS X if they haven't been configured properly to
+// be run on Linux
+
+#if !os(Linux)
+
+public protocol XCTestCaseProvider {
+    var allTests : [(String, () -> Void)] { get }
+}
+
+extension XCTestCase {
+    override public func tearDown() {
+        if let provider = self as? XCTestCaseProvider {
+            provider.assertContainsTest(invocation!.selector.description)
+        }
+
+        super.tearDown()
+    }
+}
+
+extension XCTestCaseProvider {
+    private func assertContainsTest(name: String) {
+        let contains = self.allTests.contains({ test in
+            return test.0 == name
+        })
+
+        XCTAssert(contains, "Test '\(name)' is missing from the allTests array")
+    }
+}
+
+#endif
```

Comment:
Tangential to this pull request, but we really have to solve this problem in swift-corelibs-xctest! I'm curious if it makes sense to you to have the above shims ship as part of swift-corelibs-xctest?